pdp/static/js/pdp_controls.js

```diff
@@ -171,7 +171,23 @@ function processDateRangeInput($date, fallbackFlag, $error) {
 function generateMenuTree(subtree, leafNameMapping) {
     var ul = $("<ul/>");
     /*jslint unparam: true*/
-    $.each(Object.keys(subtree), function (index, stuff) {
+
+    // elements in each subtree are arranged in alphabetical order
+    // by display name.
+    function compareDisplayName(a, b) {
+        const adn = ( leafNameMapping && a in leafNameMapping ? 
+            leafNameMapping[a] : a).toLowerCase();
+        const bdn = (leafNameMapping && b in leafNameMapping ?
+            leafNameMapping[b] : b).toLowerCase();
+        if(adn < bdn) {
+            return -1;
+        }
+        if(adn > bdn) {
+            return 1;
+        }
+        return 0;
```

Comment:
You can replace the whole set of comparisons and returns with [localeCompare](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare)

```
return adn.localeCompare(bdn);
```

Reply:
That's great. when I was typing it, I was sure there must be a better way.